src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -57,19 +57,25 @@ class TaskReplaceActor(
   def isHealthy(instance: Instance, healths: Map[Instance.Id, Seq[Health]]): Boolean = {
     val i_health = healths.find(_._1 == instance.instanceId)
     if (instance.hasConfiguredHealthChecks && i_health.isDefined)
-      return _isHealthy(i_health.get._2)
+      return _isHealthy(instance, i_health.get._2)
     else if (instance.hasConfiguredHealthChecks && !i_health.isDefined)
       return false
     else
       return instance.isRunning
   }
 
-  def _isHealthy(healths: Seq[Health]): Boolean = {
+  def _isHealthy(instance: Instance, healths: Seq[Health]): Boolean = {
     for (h <- healths)
-      if (!h.alive) return false
```

Comment:
OPT: If I understand correctly, this is more a workaround than a fix, because you "ignore" old failing healthchecks, but in fact they should not appear at all. So, I'd add a comment to say this is a workaround, not a fix, and pointing to a JIRA ticket in our backlog to address the real issue (the fact that marathon keeps old HC results) earlier in the code tree.